docs/streams/developer-guide.html

```diff
@@ -267,21 +295,532 @@
+               // Note how we change the key and the key type (similar to `selectKey`)
+               // as well as the value and the value type.
+               KStream&lt;String, Integer&gt; transformed = stream.map(
+                   (key, value) -> KeyValue.pair(value.toLowerCase(), value.length()));
+
+               // Java 7 example
+               KStream&lt;String, Integer&gt; transformed = stream.map(
+                   new KeyValueMapper&lt;byte[], String, KeyValue&lt;String, Integer&gt;&gt;() {
+                   @Override
+                   public KeyValue&lt;String, Integer&gt; apply(byte[] key, String value) {
+                       return new KeyValue&lt;&gt;(value.toLowerCase(), value.length());
+                   }
+               });
+               </pre>
+           </td>
+       </tr>
+       <tr>
+           <td><b>Map (values only)</b>: <code>KStream->KStream or KTable->KTable</code></td>
+           <td>
+               Takes one record and produces one record, while retaining the key of the original record. You can modify
+               the record value and the value type.<p>
+
+               <code>mapValues</code> is preferable to <code>map</code> because it will not cause data re-partitioning. However, it does not
+               allow you to modify the key or key type like <code>map</code> does.
+
+               <pre class="brush: java;">
+               KStream&lt;byte[], String> stream = ...;
+
+               // Java 8+ example, using lambda expressions
+               KStream&lt;byte[], String&gt; uppercased = stream.mapValues(value -> value.toUpperCase());
+
+               // Java 7 example
+               KStream&lt;byte[], String&gt; uppercased = stream.mapValues(
+                  new ValueMapper&lt;String&gt;() {
+                  @Override
+                  public String apply(String s) {
+                     return s.toUpperCase();
+                  }
+               });
+               </pre>
+           </td>
+       </tr>
+       <tr>
+           <td><b>Print</b>: <code>KStream->void or KTable->void</code></td>
+           <td>
+               Terminal operation. Prints the records to <code>System.out</code>. See Javadocs for serde and <code>toString()</code> caveats.
+
+               <pre class="brush: java;">
+               KStream&lt;byte[], String&gt; stream = ...;
+               stream.print();
+
+               // Several variants of `print` exist to e.g. override the
+               // default serdes for record keys and record values.
+               stream.print(Serdes.ByteArray(), Serdes.String());
+               </pre>
+           </td>
+       </tr>
+       <tr>
+           <td><b>SelectKey</b>: <code>KStream->KStream</code></td>
+           <td>
+               Assigns a new key, possibly of a new key type, to each record.<p>
+
+               Marks the stream for data re-partitioning: Applying a grouping or a join after <code>flatMap</code> will result in
```

Comment:
- "Marks the stream for data re-partitioning:" should be in bold.
- `flatMap` should be `selectKey` here.  (we have the same typo in the CP docs)